src/nwb/RecordingContainers.cpp

```diff
@@ -0,0 +1,65 @@
+
+#include "nwb/RecordingContainers.hpp"
+
+#include "nwb/ecephys/ElectricalSeries.hpp"
+#include "nwb/hdmf/base/Container.hpp"
+
+using namespace AQNWB::NWB;
+// Recording Container
+
+RecordingContainers::RecordingContainers() {}
+
+RecordingContainers::~RecordingContainers() {}
+
+void RecordingContainers::addData(std::unique_ptr<Container> data)
+{
+  this->containers.push_back(std::move(data));
```

Comment:
By using `std::move` and using `std::unique_ptr<Container> data` we require that ownership is being transferred from the caller to to the ``containers`` vector.  I believe this means two things:
1. the caller cannot just call `addData` with a `Container` pointer must use `std::move` 
2. the caller loses ownership of the Container and cannot access the `data` anymore

E.g.:

```
RecordingContainers rc;
std::unique_ptr<Container> myContainer = std::make_unique<MyContainer>(1);
rc.addData(std::move(myContainer));  // Must use std::move here
// myContainer is a nullptr now
```

If we are saying that `RecordingContainers` is taking on ownership of the `Container`, then I think this makes sense. Does `NWBFile` need to keep the Container it creates around or does it give up ownership. I believe it is the former (in which case this is fine), but I just wanted to confirm.

However, this transfer of ownership is a detail that may not be obvious to users, so it would be useful to document this side-effect in the docstring of the function for developers and if a user is intended to call ``addData` directly (rather than, e.g.,`NWBFile.createElectricalSeries` calling it on the users behalf), then we should also mention it in the user workflow docs. However, if it is just happening behind the scences, then mentioning it in the docsting of ``addData`` should be sufficient.